### PR TITLE
SDK update for 2023-06-01-preview

### PR DESCRIPTION
**Pull request by matthohn-msft**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/communication/Azure.ResourceManager.Communication/tests/ScenarioTests/SuppressionListTests.cs [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+            foreach (var listName in listNames)
+            {
+                var suppressionList = await CreateDefaultSuppressionListResource(_domainResource, listName);
+                resourceNames[suppressionList.Data.Name] = listName;
+            }
+
+            var suppressionLists = await _domainResource.GetSuppressionListResources().GetAllAsync().ToEnumerableAsync();
+
+            Assert.IsNotNull(suppressionLists);
+            Assert.IsTrue(suppressionLists.Count() >= listNames.Length);
```
Comment (kagbakpem) [inline sdk/communication/Azure.ResourceManager.Communication/tests/ScenarioTests/SuppressionListTests.cs:138]: Why not check for exact count?

File: sdk/communication/Azure.ResourceManager.Communication/tests/ScenarioTests/SuppressionListTests.cs [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            foreach (var address in addresses)
+            {
+                var resource = await CreateDefaultSuppressionListResource(_domainResource, address);
+                resourceNames[resource.Data.Name] = address;
+            }
+
+            var suppressionListAddresses = await suppressionList.GetSuppressionListAddressResources().GetAllAsync().ToEnumerableAsync();
+
+            Assert.IsNotNull(suppressionListAddresses);
+            Assert.IsTrue(suppressionListAddresses.Count() >= addresses.Length);
```
Comment (kagbakpem) [inline sdk/communication/Azure.ResourceManager.Communication/tests/ScenarioTests/SuppressionListTests.cs:169]: Why not check for exact count?

File: sdk/communication/Azure.ResourceManager.Communication/tests/ScenarioTests/SuppressionListTests.cs [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+                .GetSuppressionListAddressResources()
+                .CreateOrUpdateAsync(WaitUntil.Completed, created.Id.Name, data);
+
+            Assert.IsNotNull(updated);
+            Assert.AreEqual(created.Data.Id, updated.Value.Data.Id);
+            Assert.AreEqual(created.Data.Email, updated.Value.Data.Email);
+            Assert.AreNotEqual(created.Data.FirstName, updated.Value.Data.FirstName);
+            Assert.AreNotEqual(created.Data.LastName, updated.Value.Data.LastName);
+            Assert.AreEqual(created.Data.Notes, updated.Value.Data.Notes);
+        }
```
Comment (kagbakpem) [inline sdk/communication/Azure.ResourceManager.Communication/tests/ScenarioTests/SuppressionListTests.cs:209]: Can you add one check to check for the total count of the list item after remains the same as before?

File: sdk/communication/Azure.ResourceManager.Communication/tests/ScenarioTests/SuppressionListTests.cs [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    }
+}
```
Comment (kagbakpem) [inline sdk/communication/Azure.ResourceManager.Communication/tests/ScenarioTests/SuppressionListTests.cs:211]: Can you also add one test to make sure duplicates are handled properly? List of [a123@abc.com, a234@abc.com, a123@abc.com] results in only adding [a123@abc.com, a234@abc.com]?

**Comment by ArthurMa1978** [comment]
@matthohn-msft please re-record all tests by following https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md

File: sdk/communication/Azure.ResourceManager.Communication/tests/ScenarioTests/SuppressionListTests.cs (hunk 1 above)
Comment (matthohn-msft) [reply sdk/communication/Azure.ResourceManager.Communication/tests/ScenarioTests/SuppressionListTests.cs]: Updated this scenario to use exact count.

File: sdk/communication/Azure.ResourceManager.Communication/src/Generated/SuppressionListResourceData.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            ListName = listName;
+            LastUpdatedTimeStamp = lastUpdatedTimeStamp;
+            CreatedTimeStamp = createdTimeStamp;
+            DataLocation = dataLocation;
+        }
+
+        /// <summary> The the name of the suppression list. This value must match one of the valid sender usernames of the sending domain. </summary>
+        public string ListName { get; set; }
+        /// <summary> The date the resource was last updated. </summary>
+        public string LastUpdatedTimeStamp { get; }
```
Comment (archerzz) [inline sdk/communication/Azure.ResourceManager.Communication/src/Generated/SuppressionListResourceData.cs:44]: Could this be a type of `DateTimeOffset`? See an example [here](https://github.com/Azure/azure-sdk-for-net/blob/e4fce1607d65d709b19c81cc78969cd08b804244/sdk/hdinsight/Azure.ResourceManager.HDInsight/src/autorest.md?plain=1#L146).

Comment (archerzz) [reply sdk/communication/Azure.ResourceManager.Communication/src/Generated/SuppressionListResourceData.cs:44]: same as `CreatedTimeStamp` below

Comment (matthohn-msft) [reply sdk/communication/Azure.ResourceManager.Communication/src/Generated/SuppressionListResourceData.cs:44]: Please take a look at the changes.  Autorest did not detect these changes

  SuppressionListProperties.createdTimeStamp: -|date-time
  SuppressionListProperties.lastUpdatedTimeStamp: -|date-time

File: sdk/communication/Azure.ResourceManager.Communication/src/autorest.md [hunk 6]
```diff
@@ -74,6 +74,8 @@ rename-mapping:
   DnsRecord: VerificationDnsRecord
   DomainsProvisioningState: DomainProvisioningState
   ProvisioningState: CommunicationServiceProvisioningState
+  SuppressionListProperties.createdTimeStamp: -|date-time
```
Comment (archerzz) [inline sdk/communication/Azure.ResourceManager.Communication/src/autorest.md:77]:
```suggestion
  SuppressionListResource.properties.createdTimeStamp: -|date-time
```

Same as `SuppressionListProperties.lastUpdatedTimeStamp: -|date-time`

File: sdk/communication/Azure.ResourceManager.Communication/src/Generated/SuppressionListResourceData.cs (hunk 5 above)
Comment (archerzz) [reply sdk/communication/Azure.ResourceManager.Communication/src/Generated/SuppressionListResourceData.cs:44]: > Please take a look at the changes. Autorest did not detect these changes
> 
> SuppressionListProperties.createdTimeStamp: -|date-time SuppressionListProperties.lastUpdatedTimeStamp: -|date-time

@matthohn-msft It should be:

```
SuppressionListResource.properties.createdTimeStamp: -|date-time
SuppressionListResource.properties.lastUpdatedTimeStamp: -|date-time
```

I've commented in the changes in `autorest.md`.

**Review by archerzz — COMMENTED** [review]
Some codes are not up to date. Need to rebuild the codes.

**Review by archerzz — APPROVED** [review]
@matthohn-msft  The changes look fine. The remaining issues are:
- Update the codes and API signature. See the error logs for details.
- Fix the test case issues.

**Review by kagbakpem — APPROVED** [review]

**Review by archerzz — COMMENTED** [review]
@matthohn-msft You need to regen the codes, since there are some changes in the generator used in `main` branch.

**Comment by matthohn-msft** [comment]
> @matthohn-msft You need to regen the codes, since there are some changes in the generator used in `main` branch.

@archerzz code has been regenerated.

File: sdk/communication/Azure.ResourceManager.Communication/tests/ScenarioTests/SuppressionListTests.cs (hunk 4 above)
Comment (matthohn-msft) [reply sdk/communication/Azure.ResourceManager.Communication/tests/ScenarioTests/SuppressionListTests.cs:280]: Will test this with e2e tests instead of the sdk for now.

**Review by archerzz — COMMENTED** [review]
@matthohn-msft 
If you are going to release a new version, don't forget to update `CHANGELOG.md`. See [here](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process?anchor=signing-and-publishing-nuget-packges) for details.

File: sdk/communication/Azure.ResourceManager.Communication/tests/ScenarioTests/SuppressionListTests.cs [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+
+            string continuationToken = default;
+
+            await foreach (var page in collection.GetAllAsync().AsPages(continuationToken: continuationToken, pageSizeHint: pageSize))
+            {
+                pageCount++;
+
+                // // todo: investigate why pageSize is not working.
+                // Assert.AreEqual(pageSize, page.Values.Count);
+
```
[the remark this answers is not in the document]
Comment (matthohn-msft) [reply sdk/communication/Azure.ResourceManager.Communication/tests/ScenarioTests/SuppressionListTests.cs]: Determined the issue, it is in the service side. I will prepare fix @kagbakpem 
When a response has no more pages, I should be returning a null nextLink value. Because the nextLink is never null, the sdk framework will never find an end to a collection when calling GetAllAsync().

**Comment by matthohn-msft** [comment]
> @matthohn-msft 
> If you are going to release a new version, don't forget to update `CHANGELOG.md`. See [here](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process?anchor=signing-and-publishing-nuget-packges) for details.

Updates the changeling, please take a look
